cli/src/main/package/bin/ide.bat

```diff
@@ -21,10 +21,9 @@ if exist "%GIT_CORE%" (
 
 if not "%1%" == "" (
   ideasy %IDE_OPTIONS% %*
-  if not %ERRORLEVEL% == 0 (
-    echo %_fBRed%Error: IDEasy failed with exit code %ERRORLEVEL% %_RESET%
-    call :echoUseBash
-    exit /b %ERRORLEVEL%
+  if errorlevel 1 (
```

Comment:
there are various errorlevels that indicate an error. Everything except `0` is an error not just `1`.
Further, every next command (IMHO including `goto` or `echo` are new executions that update the `errorlevel`.
So did you test this with different scenarios? I still have doubts that this is working now.

Reply:
`if errorlevel <number>`" is defined as:

```
ERRORLEVEL number Specifies a true condition if the last program run
                    returned an exit code equal to or greater than the number
                    specified.
```

You can see this definition by running `if /?` in the CMD. Moreover, my observation is that `%ERRORLEVEL%` is always 0 in the first parsing of the script. When using `:goto`, it looks like the parsing is done again and the correct `%ERRORLEVEL%` is assigned to this variable.

I am still testing this behavior

Reply:
In my latest test by comparing the output of `ide` commands in git-bash and cmd/powershell, I couldn't identify any difference in behavior.

Reply:
@alfeilex thanks for testing and analysing this problem.
I am surprised that the `%ERRORLEVEL%` variable that is documented [here](https://ss64.com/nt/errorlevel.html) does not actually seem to work at all.
However, what you are doing is to ignore the error-level or exit code and always exit `ide.bat` with `0` (success).
It seems you are absolutely correct that is was always broken and your PR is not making it worse and at least fixes the error logging.
However, we need to research for the real solution. A quick search lead me to this URL that may help to fix the problem:
https://devblogs.microsoft.com/oldnewthing/20080926-00/?p=20743

Reply:
The reason why `%ERRORLEVEL%` does not work is that is it's called inside the brackets. A further solution to get the correct exit code is to call it after the `if condition`, where the `ideasy` command is executed. In that case, we would get the correct `%ERRORLEVEL%` variable and would not need subroutines or even delayed expansion. 

Would you agree? The related code would be like this:

```
if not "%1%" == "" (
  ideasy %IDE_OPTIONS% %*
)

if %ERRORLEVEL% NEQ 0 (
  echo.
  echo %_fBRed%Error: IDEasy failed with exit code %ERRORLEVEL% %_RESET%
  call :echoUseBash
  exit /b
)
```

Reply:
There is one problem with that solution. When `ide` is run without arguments, then the problem is that the `%ERRORLEVEL%` of the previous commands is used. In the first run of `ide` in a fresh terminal, this:

`if exist "%GIT_CORE%" ...` is not true and the `%ERRORLEVEL%` is 1 resulting in an output of `Error: IDEasy failed with exit code 1

Reply:
In my point of view, we have three possibilities to solve this:

- run the command isolated and use the `%ERRORLEVEL%` afterwards
- using subroutines with `goto` (current PR state)
- enable delayed expansion (`!ERRORLEVEL!`)

Reply:
Thanks for all your comments, further tests and explanations.
Indeed I was wrong with my initial comment and your approach is fine in general.

Great that you also have a solution for the remaining problem.
I fully agree that `delayed expansion` sucks and we should avoid it.
If the second option is already implemented, how can it be a fix to a problem still existing?
Actually you are free to chose the solution that seems best and easiest for your.
But just from my understanding some suggestions:
Maybe you can just use a stupid `goto` to skip the execution of `ideasy` with parameters if there are no arguments.
Then we have these executions outside parenthesis avoiding all this ugly CMD problems.

Reply:
> If the second option is already implemented, how can it be a fix to a problem still existing?

I meant that it's already implemented in the current PR implementation

> Maybe you can just use a stupid goto to skip the execution of ideasy with parameters if there are no arguments.
Then we have these executions outside parenthesis avoiding all this ugly CMD problems.

This is a nice suggestion. It have implemented this and it works fine. Thank you!

Reply:
Very nice, this looks quite clean and readable now (at least for a CMD solution) 😄